Extended LUKS encryption support (#1168)
OSInside/kiwi: doc/source/building/working_with_images/vmx_setup_for_luks.rst

.. _setup_for_luks:

KIWI Image Description Encrypted Disk
=====================================

.. sidebar:: Abstract

   This page provides further information for handling

>    vmx images built with KIWI and references the following

Reviewer: ```suggestion
   vmx images with encrypted root partitions built with KIWI and references the following
```

   articles:

   * :ref:`vmx`

A virtual disk image can be partially or fully encrypted
using the LUKS extension supported by KIWI. A fully encrypted
image also includes the data in :file:`/boot` to be encrypted.
Such an image requests the passphrase for the master key
to be entered at the bootloader stage. A partialy encrypted

> image keeps :file:`/boot` open and on an extra boot partition.

Reviewer: ```suggestion
image keeps :file:`/boot` unencrypted and on an extra boot partition.
```

Such an image requests the passphrase for the master key later
in the boot process when the root partition gets accessed by
the systemd mount service. In any case the master passphrase
is requested only once.

Update the KIWI image description as follows:

1. Software packages

>    Make sure to add the following packages to the package list

Reviewer: ```suggestion
   Make sure to add the following package to the package list:
```

   .. note::

>       Package names used in the following list matches the

Reviewer: ```suggestion
      Package names used in the following list match the
```

      package names of the SUSE distribution and might be different
      on other distributions.

   .. code:: xml

      <package name="cryptsetup"/>

2. Image Type definition

   Update the vmx image type setup as follows

>    Full Encryption Including :file:`/boot`

Reviewer: ```suggestion
   Full disk encryption including :file:`/boot`:
```

     .. code:: xml

        <type image="vmx"
            image="vmx"
            filesystem="ext4"
            bootloader="grub2"
            luks="linux"
            bootpartition="false">
        </type>

>    Partial Encryption with :file:`/boot` on an extra partition

Reviewer: ```suggestion
   Encrypted :file:`/` with an unencrypted :file:`/boot` partition:
```
